src/server/main_service.cc

```diff
@@ -958,16 +958,10 @@ optional<ErrorReply> Service::CheckKeysOwnership(const CommandId* cid, CmdArgLis
     return ErrorReply{"-CROSSSLOT Keys in request don't hash to the same slot"};
   }
 
-  // Check keys slot is in my ownership
-  const cluster::ClusterConfig* cluster_config = cluster_family_.cluster_config();
-  if (cluster_config == nullptr) {
-    return ErrorReply{kClusterNotConfigured};
-  }
-
-  if (keys_slot.has_value() && !cluster_config->IsMySlot(*keys_slot)) {
-    // See more details here: https://redis.io/docs/reference/cluster-spec/#moved-redirection
-    cluster::ClusterNodeInfo master = cluster_config->GetMasterNodeForSlot(*keys_slot);
-    return ErrorReply{absl::StrCat("-MOVED ", *keys_slot, " ", master.ip, ":", master.port)};
+  if (keys_slot.has_value()) {
+    if (auto error_str = cluster::SlotOwnershipErrorStr(*keys_slot); error_str) {
+      return ErrorReply{std::move(*error_str)};
+    }
```

Comment:
you can just return an error reply directly, will simplify the code around 🙂

Reply:
we don't need it in other places